### PR TITLE
[Term Entry] D3 Interactivity .duration()

### LINKED ISSUE #4016: [Term Entry] D3 Interactivity .duration()
### Reason/inspiration (optional)

We would like a new term entry on the `.duration()` method for [D3](https://www.codecademy.com/resources/docs/d3): Interactivity. The entry should go in a new file under `docs/content/d3/concepts/interactivity/terms/duration/duration.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```js `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added the D3 Interactivity .duration() function
- Closes #4016 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/d3/concepts/interactivity/terms/duration/duration.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  function changeColor() {
+    // Select the circle by its ID
+    d3.select('#my_circle')
+      .transition() // Start a transition
+      .duration(1000) // Set the duration to 1 second
+      .attr('fill', 'steelblue'); // Change the fill color to steelblue
+    console.log('Color changed successfully');
+  }
+</script>
+```
```
Comment (avdhoottt) [inline content/d3/concepts/interactivity/terms/duration/duration.md:66]: There's no need to add the HTML code, you can add the JavaScript part only where the method is used (Code under the Script tag).

File: content/d3/concepts/interactivity/terms/duration/duration.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Tags:
+  - 'Methods'
+  - 'Animation'
+  - 'D3'
+CatalogContent:
+  - 'learn-d3'
+  - 'paths/data-science'
+---
+
+The **`.duration()`** method sets the duration for transitions. This is crucial for creating animated visualizations, allowing developers to control how long an animation takes to move from a starting to an ending state. It accepts a single argument representing the time in milliseconds for the transition.
```
Comment (avdhoottt) [inline content/d3/concepts/interactivity/terms/duration/duration.md:16]:
```suggestion
The **`.duration()`** method sets the duration for transitions. This method is crucial for creating animated visualizations as it allows developers to control how long an animation takes to move from a starting to an ending state. It accepts a single argument representing the time in milliseconds for the transition.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@andersooi Great Entry!! I have suggested some minor changes, please make them.

**Review by avdhoottt — APPROVED** [review]
Good to go for the second review.

File: content/d3/concepts/interactivity/terms/duration/duration.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Tags:
+  - 'Methods'
+  - 'Animation'
+  - 'D3'
+CatalogContent:
+  - 'learn-d3'
+  - 'paths/data-science'
+---
+
+The **`.duration()`** method sets the duration for transitions. This method is crucial for creating animated visualizations as it allows developers to control how long an animation takes to move from a starting to an ending state. It accepts a single argument representing the time in milliseconds for the transition.
```
Comment (ishg-153) [inline content/d3/concepts/interactivity/terms/duration/duration.md:16]:
```suggestion
The **`.duration()`** method sets the time for which a transition lasts. This method is crucial for creating animated visualizations as it allows developers to control the length of an animation. It accepts a single argument representing the duration in milliseconds for the transition.
```

File: content/d3/concepts/interactivity/terms/duration/duration.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## Syntax
+
+```pseudo
+transition().duration(milliseconds)
```
Comment (ishg-153) [inline content/d3/concepts/interactivity/terms/duration/duration.md:21]:
```suggestion
transition().duration(t)
```

File: content/d3/concepts/interactivity/terms/duration/duration.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+This method requires a transition selection object and `miliseconds` as an argument. If `milliseconds` is not provided, D3 defaults to using a 250ms duration for the transition.
```
Comment (ishg-153) [inline content/d3/concepts/interactivity/terms/duration/duration.md:24]:
```suggestion
- `t`: It is the duration of the animation and the only argument taken by the `.duration()` method. If not specified, the default duration value is set to 250ms for the transition.
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@andersooi I have suggested a few changes. Please make those and let me know when you are done.

**Comment by andersooi** [comment]
Hi @ishg-153 I have made the necessary changes; bumping this just in case the review request didn't go through

**Review by ishg-153 — APPROVED** [review]
@andersooi Great! This looks ready to merge.

**Pull request merged**